usr/share/rear/backup/NETFS/default/500_make_backup.sh

```diff
@@ -71,45 +71,91 @@ LogPrint "Creating $BACKUP_PROG archive '$backuparchive'"
+            "${BACKUP_PROG_COMPRESS_OPTIONS[@]}"                                \
+            -X $TMP_DIR/backup-exclude.txt -C / -c -f -                         \
+            $(cat $TMP_DIR/backup-include.txt) $RUNTIME_LOGFILE |   \
+                                                                    \
+        $BACKUP_PROG_CRYPT_OPTIONS $BACKUP_PROG_CRYPT_KEY |         \
+                                                                    \
+        $SPLIT_COMMAND
+        pipes_rc=( ${PIPESTATUS[@]} )
+
+        # Exit code logic:
+        # - never return rc=1 (this is reserved for "tar" warning about modified files)
+        # - process exit code in pipe's reverse order
+        #   - if last command failed (e.g. "dd"), return an error
+        #   - otherwise if previous command failed (e.g. "encrypt"), return an error
+        #   ...
+        #   - otherwise return "tar" exit code
+        #
+        # When an error occurs, record the program name in $TMP_DIR/failing_backup_prog
+        # and real exit code in $TMP_DIR/failing_backup_prog_rc.
+
+        let index=${#pipes_rc[@]}-1
+        while [ $index -ge 0 ] ; do
+            rc=${pipes_rc[$index]}
+            if [ $rc -ne 0 ] ; then
+                echo "${backup_prog_shortnames[$index]}" > $TMP_DIR/failing_backup_prog
+                echo "$rc" > $TMP_DIR/failing_backup_prog_rc
+                if [ $rc -eq 1 ] && [ "${backup_prog_shortnames[$index]}" != "tar" ] ; then
+                    rc=2
+                fi
+                exit $rc
+            fi
+            # This pipe succeeded, check the previous one
+            let index--
+        done
+        # This was a success
+        exit 0
+    ;;
+    (rsync)
+        # make sure that the target is a directory
+        mkdir -p $v "$backuparchive" >&2
+        Log $BACKUP_PROG --verbose "${BACKUP_RSYNC_OPTIONS[@]}" --one-file-system --delete \
+            --exclude-from=$TMP_DIR/backup-exclude.txt --delete-excluded \
+            $(cat $TMP_DIR/backup-include.txt) "$backuparchive"
+        $BACKUP_PROG --verbose "${BACKUP_RSYNC_OPTIONS[@]}" --one-file-system --delete \
+            --exclude-from=$TMP_DIR/backup-exclude.txt --delete-excluded \
+            $(cat $TMP_DIR/backup-include.txt) "$backuparchive" >&2
+    ;;
+    (*)
+        Log "Using unsupported backup program '$BACKUP_PROG'"
+        Log $BACKUP_PROG "${BACKUP_PROG_COMPRESS_OPTIONS[@]}" \
+            $BACKUP_PROG_OPTIONS_CREATE_ARCHIVE $TMP_DIR/backup-exclude.txt \
+            "${BACKUP_PROG_OPTIONS[@]}" $backuparchive \
+            $(cat $TMP_DIR/backup-include.txt) $RUNTIME_LOGFILE > $backuparchive
+        $BACKUP_PROG "${BACKUP_PROG_COMPRESS_OPTIONS[@]}" \
+            $BACKUP_PROG_OPTIONS_CREATE_ARCHIVE $TMP_DIR/backup-exclude.txt \
+            "${BACKUP_PROG_OPTIONS[@]}" $backuparchive \
+            $(cat $TMP_DIR/backup-include.txt) $RUNTIME_LOGFILE > $backuparchive
+    ;;
 esac 2> "${TMP_DIR}/${BACKUP_PROG_ARCHIVE}.log"
```

Comment:
Ony a question from plain looking at the code:

I wonder what output that `2> "${TMP_DIR}/${BACKUP_PROG_ARCHIVE}.log"`
results in that log file when it is run in debugscripts mode via `rear -D`.

I assume then the `set -x` output is also therein (which would be the expected behaviour).

But then I woder if in the code later below using `grep` in that log file
results the actually intended lines even when there are also
tons of `set -x` lines that might (unintentionally) also match?